pkg/minikube/bootstrapper/certs.go

```diff
@@ -197,3 +218,117 @@ func generateCerts(k8s config.KubernetesConfig) error {
 
 	return nil
 }
+
+func collectCACerts() (map[string]string, error) {
+	localPath := constants.GetMinipath()
+
+	isValidPem := func(hostpath string) (bool, error) {
+		fileBytes, err := ioutil.ReadFile(hostpath)
+		if err != nil {
+			return false, err
+		}
+
+		for {
+			block, rest := pem.Decode(fileBytes)
+			if block == nil {
+				break
+			}
+
+			if block.Type == "CERTIFICATE" {
+				// certificate found
+				return true, nil
+			}
+			fileBytes = rest
+		}
+
+		return false, nil
+	}
+
+	certFiles := map[string]string{}
+
+	certsDir := filepath.Join(localPath, "certs")
+	err := filepath.Walk(certsDir, func(hostpath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info != nil && !info.IsDir() {
+			ext := strings.ToLower(filepath.Ext(hostpath))
+			if ext == ".crt" || ext == ".pem" {
+				validPem, err := isValidPem(hostpath)
+				if err != nil {
+					return err
+				}
+				if validPem {
+					filename := filepath.Base(hostpath)
+					dst := fmt.Sprintf("%s.%s", strings.TrimSuffix(filename, ext), "pem")
+					certFiles[hostpath] = path.Join(constants.CACertificatesDir, dst)
+				}
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "provisioning: traversal certificates dir %s", certsDir)
+	}
+
+	for _, excluded := range []string{"ca.pem", "cert.pem"} {
+		certFiles[filepath.Join(certsDir, excluded)] = ""
+	}
+
+	// populates minikube CA
+	certFiles[filepath.Join(localPath, "ca.crt")] = path.Join(constants.CACertificatesDir, "minikubeCA.pem")
+
+	filtered := map[string]string{}
+	for k, v := range certFiles {
+		if v != "" {
+			filtered[k] = v
+		}
+	}
+	return filtered, nil
+}
+
+func configureCACerts(cmd command.Runner, caCerts map[string]string) error {
+	getSubjectHash := func(hostpath string) (string, error) {
+		out, err := cmd.CombinedOutput(fmt.Sprintf("openssl x509 -hash -noout -in '%s'", hostpath))
+		if err != nil {
+			return "", err
+		}
+
+		stringHash := strings.ReplaceAll(out, "\n", "")
```

Comment:
Would it be possible to use `strings.TrimSpace()`?

Reply:
Will use `strings.TrimSpace()` instead.